src/main.rs

```diff
@@ -240,8 +250,54 @@ mod tests {
         let args = "cargo clippy -Zunstable-options"
             .split_whitespace()
             .map(ToString::to_string);
-        let cmd = ClippyCmd::new(args);
+        let cmd = ClippyCmd::new(args, None);
+
         assert_eq!("check", cmd.cargo_subcommand);
         assert_eq!("RUSTC_WORKSPACE_WRAPPER", cmd.path_env());
     }
+
+    #[test]
+    fn clippy_args_into_rustflags() {
+        let args = "cargo clippy -- -W clippy::as_conversions"
+            .split_whitespace()
+            .map(ToString::to_string);
+        let rustflags = None;
+        let cmd = ClippyCmd::new(args, rustflags);
+
+        assert_eq!("-W clippy::as_conversions", cmd.rustflags.unwrap());
+    }
+
+    #[test]
+    fn clippy_args_respect_existing_rustflags() {
+        let args = "cargo clippy -- -D clippy::await_holding_lock"
+            .split_whitespace()
+            .map(ToString::to_string);
+        let rustflags = Some(r#"--cfg feature="some_feat""#.into());
+        let cmd = ClippyCmd::new(args, rustflags);
+
+        assert_eq!(
+            r#"-D clippy::await_holding_lock --cfg feature="some_feat""#,
+            cmd.rustflags.unwrap()
+        );
+    }
+
+    #[test]
+    fn no_env_change_if_no_clippy_args() {
+        let args = "cargo clippy".split_whitespace().map(ToString::to_string);
+        let rustflags = Some(r#"--cfg feature="some_feat""#.into());
+        let cmd = ClippyCmd::new(args, rustflags);
+
+        assert!(cmd.clippy_args.is_none());
+        assert!(cmd.rustflags.is_none());
```

Comment:
Hm, shouldn't this be `r#"--cfg feature="some_feat""#`? `ClippyCmd` shouldn't remove rustflags or am I missing something?

Reply:
It is my understanding that the child process (`cargo check` in this case) will inherit the environment from the parent (`cargo clippy`), so not setting the var here should leave the existing value intact.

Reply:
Ah, but that only works, because we don't override the original `rustflags`, when ``cmd.rustflags.is_none() == true``. I think that makes the code too confusing and I would rather deal with that in the ``into_std_command`` method.

Reply:
Not sure if I addressed this comment in my last commit. Do you prefer to re-set the `RUSTFLAGS` even if we don't change them?

Reply:
Nope just passing it through in that case is fine. I just wanted to have the handling of the rustflags in a single place, instead of splitting it up.